### ISSUE TITLE
[Bug] Network Costs is not showing any cross-zone traffic,

### ISSUE THREAD

**Issue by AaronAllBright** [kubecost/features-bugs#33]
### Kubecost Version

1.107.1 (f87c784)

### Kubernetes Version

1.28

### Kubernetes Platform

EKS

### Description

I have a reasonably large multi-az cluster with KubeCost installed. Network costs daemon is installed and is up and running. No cross-zone traffic is shown in the KubeCost UI (Allocation / Network Costs). There is a large "Adjustment" cost, implying that KubeCost is aware of the spend, but just can't classify the traffic.

I have followed instructions at [Network Cost Configuration - Troubleshooting](https://docs.kubecost.com/install-and-configure/advanced-configuration/network-costs-configuration#troubleshooting)

I think I may have discovered a clue, and I'll include screenshots below. It seems that the grafana dashboard for network costs is also missing cross zone and cross region costs, but when I look at the promql that's fetching the metrics, it's clearly wrong (see screenshots below). It's using the wrong labels: `sameRegion` instead of `same_region` and sameZone instead of `same_zone`. Is it possible the cost model engine is also looking at the wrong labels?

As you can see from the screenshots and config below, the classified network traffic metrics are there in prometheus, but not making it all the way to the UI somehow.

Any help would be greatly appreciated!

### Steps to reproduce

1. Configure KubeCost in a multi-az EKS cluster
2. Enable network costs
3. Generate some traffic
4. Look at the Allocations / Network Costs section of the UI for any namespace
5. Note that while traffic is captured, and costs are reconciled with AWS, the cross-zone and cross-region columns are all zeros.

### Expected behavior

I would expect to see some traffic in the cross-zone and cross-region columns of the Allocation / Network Costs screen.

### Impact

One of our main use cases for KubeCost is to help tease out cross zone network costs. Since it's not classifying properly, this is making the tool much less useful.

### Screenshots

Here you can see all Prometheus kubecost-networking targets are up:
![image](https://github.com/kubecost/features-bugs/assets/47489965/5554ec67-5b29-42d4-afbe-a60a0f5cdc82)

Here you can see that `kubecost_pod_network_egress_bytes_total` metrics are being captured:
![image](https://github.com/kubecost/features-bugs/assets/47489965/2c4258ff-c9d7-409f-bfa9-f078d86bbe12)

Here you can see that cross zone and cross region traffic is missing:
![image](https://github.com/kubecost/features-bugs/assets/47489965/e9445c8c-56ac-4a0d-a2c9-66e04e523fbe)

And here you can see why. The name of the labels being used in the promql is wrong:
![image](https://github.com/kubecost/features-bugs/assets/47489965/4c563955-d298-49cb-8290-8dc11154795c)

If I fix it, you can see that the data appears. This proves that my classification rules are working. The kubecost-networking daemon is classifying traffic.
![image](https://github.com/kubecost/features-bugs/assets/47489965/2499785a-cced-41b7-a533-a750146f25ca)

I have included my network costs config below.

Here's what things look like in the UI. This is network costs for kube-system namespace:
![image](https://github.com/kubecost/features-bugs/assets/47489965/ae9f2b5a-0c22-4b66-b40b-e5ad79aa7b47)

### Logs

```Shell
Here you can see all Kubecost services and pods up and running:
$ k get all --namespace kubecost
NAME                                              READY   STATUS    RESTARTS   AGE
pod/kubecost-cost-analyzer-8779958fd-978f6        2/2     Running   0          29h
pod/kubecost-grafana-79c8884f54-gzrv9             2/2     Running   0          7d4h
pod/kubecost-network-costs-2mfm2                  1/1     Running   0          7d4h
pod/kubecost-network-costs-2q645                  1/1     Running   0          7d4h
pod/kubecost-network-costs-59x2p                  1/1     Running   0          7d4h
pod/kubecost-network-costs-5jv2b                  1/1     Running   0          7d4h
pod/kubecost-network-costs-5mrpb                  1/1     Running   0          7d4h
pod/kubecost-network-costs-5vd7s                  1/1     Running   0          7d4h
pod/kubecost-network-costs-8plj7                  1/1     Running   0          7d4h
pod/kubecost-network-costs-8t4mx                  1/1     Running   0          7d4h
pod/kubecost-network-costs-bfvxp                  1/1     Running   0          7d4h
pod/kubecost-network-costs-bj49l                  1/1     Running   0          7d4h
pod/kubecost-network-costs-bpmjf                  1/1     Running   0          7d4h
pod/kubecost-network-costs-bvnkn                  1/1     Running   0          7d4h
pod/kubecost-network-costs-cgvnd                  1/1     Running   0          7d4h
pod/kubecost-network-costs-d5gh6                  1/1     Running   0          7d4h
pod/kubecost-network-costs-dfftx                  1/1     Running   0          7d4h
pod/kubecost-network-costs-dl9gx                  1/1     Running   0          7d4h
pod/kubecost-network-costs-dmbr4                  1/1     Running   0          7d4h
pod/kubecost-network-costs-dv86s                  1/1     Running   0          7d4h
pod/kubecost-network-costs-dvxgv                  1/1     Running   0          7d4h
pod/kubecost-network-costs-f9qnk                  1/1     Running   0          7d4h
pod/kubecost-network-costs-gd46d                  1/1     Running   0          7d4h
pod/kubecost-network-costs-h6vlp                  1/1     Running   0          7d4h
pod/kubecost-network-costs-hwkww                  1/1     Running   0          7d4h
pod/kubecost-network-costs-jjhcc                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kbhhw                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kcxbf                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kdgns                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kllst                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kpts6                  1/1     Running   0          7d4h
pod/kubecost-network-costs-kqkwp                  1/1     Running   0          7d4h
pod/kubecost-network-costs-mc8v2                  1/1     Running   0          7d4h
pod/kubecost-network-costs-n9rd5                  1/1     Running   0          7d4h
pod/kubecost-network-costs-nl7dc                  1/1     Running   0          7d4h
pod/kubecost-network-costs-nsn9r                  1/1     Running   0          6d2h
pod/kubecost-network-costs-q66fg                  1/1     Running   0          7d4h
pod/kubecost-network-costs-qkvt5                  1/1     Running   0          7d4h
pod/kubecost-network-costs-qn85p                  1/1     Running   0          7d4h
pod/kubecost-network-costs-r9s69                  1/1     Running   0          7d4h
pod/kubecost-network-costs-rlmzj                  1/1     Running   0          7d4h
pod/kubecost-network-costs-rms9h                  1/1     Running   0          7d4h
pod/kubecost-network-costs-rxpj9                  1/1     Running   0          7d4h
pod/kubecost-network-costs-sqpxp                  1/1     Running   0          7d4h
pod/kubecost-network-costs-t6tdh                  1/1     Running   0          7d4h
pod/kubecost-network-costs-tl58d                  1/1     Running   0          7d4h
pod/kubecost-network-costs-twfxk                  1/1     Running   0          7d4h
pod/kubecost-network-costs-tzkqc                  1/1     Running   0          7d4h
pod/kubecost-network-costs-v546l                  1/1     Running   0          7d4h
pod/kubecost-network-costs-w9h4d                  1/1     Running   0          7d4h
pod/kubecost-network-costs-w9lps                  1/1     Running   0          7d4h
pod/kubecost-network-costs-wfcq2                  1/1     Running   0          7d4h
pod/kubecost-network-costs-wkzmt                  1/1     Running   0          7d4h
pod/kubecost-network-costs-wxgjh                  1/1     Running   0          7d4h
pod/kubecost-network-costs-wzhzv                  1/1     Running   0          7d4h
pod/kubecost-network-costs-zcgb4                  1/1     Running   0          7d4h
pod/kubecost-prometheus-server-74b9f65cb5-gl97j   2/2     Running   0          7d4h

NAME                                 TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
service/kubecost-cost-analyzer       ClusterIP   172.20.127.37   <none>        9003/TCP,9090/TCP   7d4h
service/kubecost-grafana             ClusterIP   172.20.184.57   <none>        80/TCP              7d4h
service/kubecost-network-costs       ClusterIP   None            <none>        3001/TCP            7d4h
service/kubecost-prometheus-server   ClusterIP   172.20.46.173   <none>        80/TCP              7d4h

NAME                                    DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/kubecost-network-costs   54        54        54      54           54          <none>          7d4h

NAME                                         READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kubecost-cost-analyzer       1/1     1            1           7d4h
deployment.apps/kubecost-grafana             1/1     1            1           7d4h
deployment.apps/kubecost-prometheus-server   1/1     1            1           7d4h

NAME                                                    DESIRED   CURRENT   READY   AGE
replicaset.apps/kubecost-cost-analyzer-675b4d75f5       0         0         0       6d3h
replicaset.apps/kubecost-cost-analyzer-68d47798d8       0         0         0       7d4h
replicaset.apps/kubecost-cost-analyzer-76fd85c97d       0         0         0       6d3h
replicaset.apps/kubecost-cost-analyzer-8779958fd        1         1         1       29h
replicaset.apps/kubecost-cost-analyzer-fff47d548        0         0         0       6d3h
replicaset.apps/kubecost-grafana-79c8884f54             1         1         1       7d4h
replicaset.apps/kubecost-prometheus-server-74b9f65cb5   1         1         1       7d4h

And here is the network costs config:

$ k get cm network-costs-config -o yaml
apiVersion: v1
data:
  config.yaml: |
    destinations:
      cross-region: []
      direct-classification:
      - ips:
        - 10.0.3.0/24
        - 10.0.60.0/22
        - 10.0.72.0/22
        - 10.0.80.0/24
        region: us-east1
        zone: us-east1-a
      - ips:
        - 10.0.4.0/24
        - 10.0.68.0/22
        - 10.0.76.0/22
        - 10.0.81.0/24
        region: us-east1
        zone: us-east1-b
      - ips:
        - 10.0.5.0/24
        - 10.0.36.0/22
        - 10.0.82.0/24
        region: us-east1
        zone: us-east1-c
      - ips:
        - 10.0.32.0/22
        - 10.0.56.0/22
        - 10.0.83.0/24
        - 10.0.92.0/24
        region: us-east1
        zone: us-east1-d
      - ips:
        - 10.0.93.0/24
        region: us-east1
        zone: us-east1-e
      - ips:
        - 10.0.40.0/22
        - 10.0.64.0/22
        - 10.0.84.0/24
        - 10.0.94.0/24
        region: us-east1
        zone: us-east1-f
      in-region: []
      in-zone:
      - 127.0.0.0/8
      - 169.254.0.0/16
      - 172.16.0.0/12
      - 192.168.0.0/16
      internet: []
    services:
      amazon-web-services: true
      azure-cloud-services: false
      google-cloud-services: false
kind: ConfigMap
metadata:
  annotations:
    meta.helm.sh/release-name: kubecost
    meta.helm.sh/release-namespace: kubecost
  creationTimestamp: "2023-12-05T14:39:35Z"
  labels:
    app: cost-analyzer
    app.kubernetes.io/instance: kubecost
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: cost-analyzer
    helm.sh/chart: cost-analyzer-1.107.1
  name: network-costs-config
  namespace: kubecost
  resourceVersion: "2931042379"
  uid: d4064385-5f4c-402c-8505-206fc6b89cb1
```

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/features-bugs/blob/main/ISSUE_GUIDELINES.md) and this is a bug impacting only the Kubecost application.
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by dwbrown2** [comment]
Thank you for the detailed report, @AaronAllBright. Our team is finalizing a release and will get back as soon as possible. That said, @mbolt35 do you have any preliminary insights?

**Comment by AjayTripathy** [comment]
Actually @mbolt35 the daemonset looks appropriate. This feels like an issue with the collection of this data and display. I believe this should be fixed in 2.0 with @avrodrigues5 's improvements, but pinging him into this to check.

**Comment by mbolt35** [comment]
OK, this problem had a pretty far reaching blast radius. I had somehow missed Alan's changes to fix this, or it would've been more obvious sooner. This issue points out the problem explicitly: https://github.com/kubecost/features-bugs/issues/45

It is related to the move to rust network costs. The prom metric writer insisted on writing _ style labels instead of the pascal casing that was used in the go version. Alan's change updates the queries to use the correct labels in KCM, but opencost is still using the camelCase labels.

For compatibility reasons, I would suggest that we update the network costs image to use the camel case label names and revert Alan's changes. Since we have been on the rust version in the helm chart for quite some time now, this might create some holes in the network metrics when we change back. The other alternative is that we don't change the image or Alan's changes, and we update opencost's query labels instead. 

@AjayTripathy would love your opinion here. Is there some prom black magic we can leverage to support both? Because it involves the metrics, this is just a really nasty issue - apologies for not catching it earlier :-(

**Comment by AjayTripathy** [comment]
I believe we will want to merge #2478 for 2.0 cc @teevans

**Comment by AaronAllBright** [comment]
I can confirm this appears to be fixed in 2.0. Thank you!

**Comment by AjayTripathy** [comment]
Closing!
